sklearn/cluster/tests/test_k_means.py

```diff
@@ -1276,3 +1276,16 @@ def test_predict_does_not_change_cluster_centers(is_sparse):
 
     y_pred2 = kmeans.predict(X)
     assert_array_equal(y_pred1, y_pred2)
+
+
+@pytest.mark.parametrize("Estimator", [KMeans, MiniBatchKMeans])
+def test_sample_weight_init(Estimator):
+    """Check that sample weight is used during init."""
+    rng = np.random.RandomState(0)
+    X, _ = make_blobs(n_samples=200, n_features=10, centers=10, random_state=0)
+    sample_weight = rng.uniform(size=200)
+    kmeans = Estimator(random_state=0, init="random")
+    y_pred_sw = kmeans.fit(X, sample_weight=sample_weight).cluster_centers_
+    kmeans = Estimator(random_state=0, init="random")
+    y_pred = kmeans.fit(X).cluster_centers_
+    assert (y_pred_sw != y_pred).any()
```

Comment:
It might be worth to have the global random seed to check that the test is not sensitive to it:

```suggestion
def test_sample_weight_init(Estimator, global_random_seed):
    """Check that sample weight is used during init."""
    rng = np.random.RandomState(global_random_seed)
    X, _ = make_blobs(n_samples=200, n_features=10, centers=10, random_state=global_random_seed)
    sample_weight = rng.uniform(size=200)
    kmeans = Estimator(random_state=global_random_seed, init="random")
    y_pred_sw = kmeans.fit(X, sample_weight=sample_weight).cluster_centers_
    kmeans = Estimator(random_state=global_random_seed, init="random")
    y_pred = kmeans.fit(X).cluster_centers_
    assert (y_pred_sw != y_pred).any()
```